account_banking_mandate/models/res_partner_bank.py

```diff
@@ -15,14 +15,14 @@ class ResPartnerBank(models.Model):
         help='Banking mandates represent an authorization that the bank '
              'account owner gives to a company for a specific operation.')
 
-    @api.constrains('company_id', 'mandate_ids')
```

Comment:
Why removing the condition when a new mandate is created?

Reply:
This constraint is checking that the user doesn't change the company of the res.partner.bank causing incongruencies. If mandates are created doesn't affect this.

Reply:
Yes, it affects as well, as you can introduce a mandate that doesn't follow the rules. Think twice about this.

Reply:
There's already a line that checks this when creating a mandate: https://github.com/OCA/bank-payment/blob/11.0/account_banking_mandate/models/account_banking_mandate.py#L84

Reply:
Then you can remove that line and let this one. It's redundant. KISS principle.

Reply:
I don't think so. This applies only when creating a mandate through a res.partner.bank, but if you were to create a mandate from scratch then this code wouldn't be executed.

Reply:
Are you sure. Please try, as I think it's also applied. Take into account that the source doesn't matter. What determines if something has changed is the one2many in the model, which is always updated.

Reply:
@rgarnau is correct, if you remove the constraint from mandate it allows you to add a mandate that has a different company.

If you maintain the 'mandate_ids' in the constraint, and erase the 'company_id' constraint from mandate (I mean the one that compares mandate and bank account companies) then you can create a mandate directly with not the same company.
If you have the company constraint in the mandate, even if you change the mandate_ids, it will raise first the one in the mandate.

![fireshot capture 16 - fr66 1212 1212 1212 1212 1212 121 - od_ - http___localhost_9169_web id 11 vi](https://user-images.githubusercontent.com/2323616/36140160-8a68e3f2-10a0-11e8-85a0-893e59ac1c35.png)
You can see in the picture how I was able to have mandates with different companies.